# dev-python/certifi: cross compilation is broken, importing its successor.
# https://bugs.gentoo.org/734988

diff --git a/setup.py b/setup.py
index 294dbfb..47bb069 100755
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,17 @@
 #!/usr/bin/env python
 # vim:fileencoding=utf-8:noet

-from setuptools import setup
+# While I generally consider it an antipattern to try and support both
+# setuptools and distutils with a single setup.py, in this specific instance
+# where certifi is a dependency of setuptools, it can create a circular
+# dependency when projects attempt to unbundle stuff from setuptools and pip.
+# Though we don't really support that, it makes things easier if we do this and
+# should hopefully cause less issues for end users.
+
+try:
+    from setuptools import setup
+except ImportError:
+    from distutils.core import setup

 from certifi import __version__
